core/xim/CmdProcessor.lua

```diff
@@ -67,40 +66,40 @@ function CmdProcessor:install(disable_info)
         if not disable_info then self._pm_executor:info(xpkg) end
 
         if is_installed then
-            cprint("[xlings:xim]: already installed - ${green bright}%s${clear}", self.target.name)
-            index_manager.status_changed_pkg[self._target] = {installed = true}
+            cprint("[xlings:xim]: already installed - ${green bright}%s${clear}", self.target)
+            index_manager.status_changed_pkg[self.target] = {installed = true}
         else
             -- please input y or n
             if self.cmds.yes ~= true then
-                local msg = string.format("${bright}install %s? (y/n)", self.target.name)
+                local msg = string.format("${bright}install %s? (y/n)", self.target)
```

Comment:
```lua
if not self._pm_executor:support() then
    cprint("[xlings:xim]: ${red}<%s>-platform not support${clear} - ${green}%s${clear}", os.host(), self.target)
    return
end

local xpkg = self.target
local is_installed = self._pm_executor:installed(xpkg)

if not disabled_info then
    ...
end

if is_installed then
    print(...)
end

if not (--prompt logic, your prompt design is bad, because you still need to check cond outside your function,that do nothing) then
     return
end

-- deps logic

-- install-failure logic


```

Reply:
There's same problem about `remove` logic, you could refactor that.